pkg/noderesourcetopology/config.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"k8s.io/klog/v2"
+
+	topologyv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+)
+
+const (
+	AttributeScope  = "topologyManagerScope"
+	AttributePolicy = "topologyManagerPolicy"
+)
+
+// TODO: can we reuse constants from k/k/pkg/kubelet/cm/topology_manager ?
+const (
+	ScopeContainer = "container"
+	ScopePod       = "pod"
+)
+
+// TODO: can we reuse constants from k/k/pkg/kubelet/cm/topology_manager ?
+const (
+	PolicyNone           = "none"
+	PolicyBestEffort     = "best-effort"
+	PolicyRestricted     = "restricted"
+	PolicySingleNUMANode = "single-numa-node"
+)
+
+// TODO: handle topologyManagerPolicyOptions added in k8s 1.26
+
+func IsValidScope(scope string) bool {
+	if scope == ScopeContainer || scope == ScopePod {
+		return true
+	}
+	return false
+}
+
+func IsValidPolicy(policy string) bool {
+	if policy == PolicyNone || policy == PolicyBestEffort || policy == PolicyRestricted || policy == PolicySingleNUMANode {
+		return true
+	}
+	return false
+}
+
+type TopologyManagerConfig struct {
+	Scope  string
+	Policy string
+}
+
+func makeTopologyManagerConfigDefaults() TopologyManagerConfig {
+	return TopologyManagerConfig{
+		Scope:  ScopeContainer,
+		Policy: PolicyNone,
+	}
+}
+
+func topologyManagerConfigFromNodeResourceTopology(nodeTopology *topologyv1alpha2.NodeResourceTopology) TopologyManagerConfig {
+	conf := makeTopologyManagerConfigDefaults()
+	// backward compatibility (v1alpha2 and previous)
+	updateTopologyManagerConfigFromTopologyPolicies(&conf, nodeTopology.Name, nodeTopology.TopologyPolicies)
```

Comment:
IIUC, we need both these ways of obtaining topology policies as the exporter still might be using the `topologyPolicies` field for exposing information and we want to continue supporting that as  a precautionary measure. We are intentionally taking the route of soft deprecation and not fully deprecating the use of `topologyPolicies` in the v1alpha2 version of NRT API . Let's emphasize this and the fact that in the next iteration (when we move to v1beta1 version) we would be deprecating the use of `topologyPolicies` field.

Reply:
Right. Added a code comment *AND* a warning